### DISCUSSION TITLE
How to change manually the learning rate during the training

### DISCUSSION THREAD

**Discussion by farahFif** [Lightning-AI/pytorch-lightning#7169] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
I want to change the learning rate after some epochs ( since I am loading another model) . I want to do it manually , I tried by calling configure_optimizer hook the learning rate changed but the scheduler is no longer working.

**Comment by awaelchli** [comment]
Hi
You do it as you would in pytorch (I assume you want to do it without a scheduler): 

```python
for group in optimizer.param_groups:
    group["lr"] = 0.12345
```
You can access the optimizer through the trainer (returns one or multiple):
`optimizer = trainer.optimizers()`

**Reply by farahFif** [reply]
Hi , indeed the learning rate changes, however the scheduler is no longer working . To be precise in the hook on_train_epoch_start I am puting a condition then change the LR. The learning changes to the specified value but it never changes after that. I am using OneCycleLR scheduler

```python
    def on_train_epoch_start(self) -> None:
       ... some code here ..
       if self.current_epoch  == 5:
             for group in self.optimizer.param_groups:
                 group["lr"] = 0.01
```

**Reply by awaelchli** [reply]
You did not mention that you are already using a LR scheduler in your first post. In that case, it's probably best if you implement your own scheduler. For example, subclass [OnceCycleLR](https://github.com/pytorch/pytorch/blob/45692fbef0992f7d46c60f4e6d508bbf061db0f8/torch/optim/lr_scheduler.py#L1049) and override the get_lr() method with your new rule.
